### PR TITLE
Cppunit fix

### PR DESCRIPTION
**Pull request by extrowerk**
Still not sure about it. Review required...
Closes https://github.com/haikuports/haikuports/issues/2453

### CODE REVIEW

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 1]
```diff
@@ -11,27 +10,30 @@ COPYRIGHT="2000 Jerome Lacoste
...
 LICENSE="GNU LGPL v2.1"
-REVISION="1"
+REVISION="2"
 SOURCE_URI="http://dev-www.libreoffice.org/src/cppunit-$portVersion.tar.gz"
 CHECKSUM_SHA256="3d569869d27b48860210c758c4f313082103a5e58219a7669b52bfd29d674780"
 
 ARCHITECTURES="x86 x86_gcc2 x86_64"
 SECONDARY_ARCHITECTURES="x86"
 
+libVersion=$portVersion
```
Comment (fbrosson) [inline dev-util/cppunit/cppunit-1.14.0.recipe]: This is what I would do:
```diff
-libVersion=$portVersion
+libVersion="0.0.0"
-libVersionCompat="$libVersion compat >= ${libVersion%.*}" 
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+portVers="${portVersion%.*}"
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 2]
```diff
@@ -11,27 +10,30 @@ COPYRIGHT="2000 Jerome Lacoste
...
+
 PROVIDES="
 	cppunit$secondaryArchSuffix = $portVersion
-	cmd:cppunit_config$secondaryArchSuffix = $portVersion compat >= 1.14
-	cmd:DllPlugInTester$secondaryArchSuffix = $portVersion compat >= 1.14
-	lib:libcppunit$secondaryArchSuffix = $portVersion compat >= 1.14
-	lib:libcppunit_1.14$secondaryArchSuffix = $portVersion compat >= 1.14
+	cmd:cppunit_config$secondaryArchSuffix = $libVersionCompat
+	cmd:DllPlugInTester$secondaryArchSuffix = $libVersionCompat
+	lib:libcppunit_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
```
Comment (fbrosson) [inline dev-util/cppunit/cppunit-1.14.0.recipe]:
```diff
-	cmd:cppunit_config$secondaryArchSuffix = $libVersionCompat
+	cmd:cppunit_config$secondaryArchSuffix = $portVersion
-	cmd:DllPlugInTester$secondaryArchSuffix = $libVersionCompat
+	cmd:DllPlugInTester$secondaryArchSuffix = $portVersion
-	lib:libcppunit_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	lib:libcppunit_$portVers$secondaryArchSuffix = $libVersionCompat
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 3]
```diff
@@ -11,27 +10,30 @@ COPYRIGHT="2000 Jerome Lacoste
...
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
 	"
 
 PROVIDES_devel="
 	cppunit${secondaryArchSuffix}_devel = $portVersion
-	devel:libcppunit$secondaryArchSuffix = $portVersion compat >= 1.14
+	devel:libcppunit$secondaryArchSuffix = $libVersionCompat
+	devel:libcppunit_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
```
Comment (fbrosson) [inline dev-util/cppunit/cppunit-1.14.0.recipe]:
```diff
-	devel:libcppunit_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	devel:libcppunit_$portVers$secondaryArchSuffix = $libVersionCompat
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 4]
```diff
@@ -50,6 +52,9 @@ BUILD_PREREQUIRES="
 	cmd:sed
 	"
 
+defineDebugInfoPackage cppunit$secondaryArchSuffix \
+	$libDir/libcppunit-${libVersion%.*}.so.0.0.0
```
Comment (fbrosson) [inline dev-util/cppunit/cppunit-1.14.0.recipe]:
```diff
-	$libDir/libcppunit-${libVersion%.*}.so.0.0.0
+	$libDir/libcppunit-$portVers.so.$libVersion
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 5]
```diff
@@ -64,7 +69,7 @@ INSTALL()
 
 	rm $libDir/libcppunit.la
 
-	prepareInstalledDevelLib libcppunit
+	prepareInstalledDevelLibs libcppunit-${libVersion%.*} libcppunit
```
Comment (fbrosson) [inline dev-util/cppunit/cppunit-1.14.0.recipe]:
```diff
-	prepareInstalledDevelLibs libcppunit-${libVersion%.*} libcppunit
+	prepareInstalledDevelLib libcppunit-$portVers
```

File: dev-util/cppunit/cppunit-1.14.0.recipe (hunk 2 above)
Comment (fbrosson) [reply dev-util/cppunit/cppunit-1.14.0.recipe]: Or maybe:
```diff
-	cmd:cppunit_config$secondaryArchSuffix = $libVersionCompat
+	cmd:cppunit_config$secondaryArchSuffix = $portVersionCompat
-	cmd:DllPlugInTester$secondaryArchSuffix = $libVersionCompat
+	cmd:DllPlugInTester$secondaryArchSuffix = $portVersionCompat
-	lib:libcppunit_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	lib:libcppunit_$portVers$secondaryArchSuffix = $libVersionCompat
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 6]
```diff
@@ -50,6 +55,9 @@ BUILD_PREREQUIRES="
 	cmd:sed
 	"
 
+defineDebugInfoPackage cppunit$secondaryArchSuffix \
+	$libDir/libcppunit-$portVers.so.$libVersion
+
 BUILD()
 {
 	libtoolize --force --copy --install
```
Comment (alaviss) [inline dev-util/cppunit/cppunit-1.14.0.recipe]: `autoreconf` would call `libtoolize` automatically if needed, so this line is not required

```diff
-	libtoolize --force --copy --install
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 7]
```diff
@@ -50,6 +55,9 @@ BUILD_PREREQUIRES="
 	cmd:sed
```
Comment (alaviss) [inline dev-util/cppunit/cppunit-1.14.0.recipe]: Can you please sort `BUILD_PREREQUIRES`?

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 8]
```diff
@@ -1,7 +1,6 @@
 SUMMARY="C++ unit testing framework"
-DESCRIPTION="
-CppUnit is a C++ unit testing framework. It started its life as a port of \
-JUnit to C++ by Michael Feathers."
+DESCRIPTION="CppUnit is a C++ unit testing framework. It started its life as \
+a port of JUnit to C++ by Michael Feathers."
 HOMEPAGE="http://cppunit.sourceforge.net/"
```
Comment (alaviss) [inline dev-util/cppunit/cppunit-1.14.0.recipe]: The linked `HOMEPAGE` only contains a link to some "FrontPage" which links to an empty page.

```diff
-HOMEPAGE="http://cppunit.sourceforge.net/"
+HOMEPAGE="https://www.freedesktop.org/wiki/Software/cppunit/"
```

File: dev-util/cppunit/cppunit-1.14.0.recipe [hunk 9]
```diff
@@ -11,27 +10,33 @@ COPYRIGHT="2000 Jerome Lacoste
 	2001-2002, 2007-2008 Steve Robbins
 	2012-2013 Harvey Brydon"
 LICENSE="GNU LGPL v2.1"
-REVISION="1"
+REVISION="2"
 SOURCE_URI="http://dev-www.libreoffice.org/src/cppunit-$portVersion.tar.gz"
```
Comment (alaviss) [inline dev-util/cppunit/cppunit-1.14.0.recipe]:
```diff
-SOURCE_URI="http://dev-www.libreoffice.org/src/cppunit-$portVersion.tar.gz"
+SOURCE_URI="http://dev-www.libreoffice.org/src/cppunit-$portVersion.tar.gz"
```

Comment (fbrosson) [reply dev-util/cppunit/cppunit-1.14.0.recipe]: @alaviss, I guess you meant:
```diff
-SOURCE_URI="http://dev-www.libreoffice.org/src/cppunit-$portVersion.tar.gz"
+SOURCE_URI="https://dev-www.libreoffice.org/src/cppunit-$portVersion.tar.gz"
```
(You forgot to add the "s" after the copy paste.)

**Pull request merged**